FoodPin/FoodPin/Controller/WalkthroughPageViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  WalkthroughPageViewController.swift
+//  FoodPin
+//
+//  Created by Andreea Sburlea on 11.05.2022.
+//
+
+import UIKit
+protocol WalkthroughPageViewControllerDelegate: AnyObject {
+    func didUpdatePageIndex(currentIndex: Int)
+}
+
+class WalkthroughPageViewController: UIPageViewController {
+    // swiftlint:disable line_length
+    // swiftlint:disable force_cast
+
+    var pageHeadings = ["CREATE YOUR OWN FOOD GUIDE", "SHOW YOU THE LOCATION", "DISCOVER GREAT RESTAURANTS"]
+    var pageImages = ["onboarding-1", "onboarding-2", "onboarding-3"]
+    var pageSubHeadings = ["Pin your favorite restaurants and create your own food guide",
+                           "Search and locate your favorite restaurant on Maps",
+                           "Find restaurants shared by your friends and other foodies"]
+    var currentIndex = 0
+
+    weak var walkthroughDelegate: WalkthroughPageViewControllerDelegate?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        delegate = self
+
+        // Set the data source to itself
+        dataSource = self
+
+        // Create the first walkthrough screen
+        if let startingViewController = contentViewController(at: 0) {
+            setViewControllers([startingViewController], direction: .forward, animated: true, completion: nil)
+        }
+    }
+}
+
+extension WalkthroughPageViewController: UIPageViewControllerDataSource {
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        var index = (viewController as! WalkthroughContentViewController).index
```

Comment:
Here you can add an if let statement instead of force unwrapping the viewController